src/components/KYCWall/BaseKYCWall.tsx

```diff
@@ -0,0 +1,280 @@
+import _ from 'lodash';
+import React, {SyntheticEvent, useCallback, useEffect, useRef, useState} from 'react';
+import {Dimensions, EmitterSubscription, NativeTouchEvent} from 'react-native';
+import {OnyxEntry, withOnyx} from 'react-native-onyx';
+import {ValueOf} from 'type-fest';
+import AddPaymentMethodMenu from '@components/AddPaymentMethodMenu';
+import * as BankAccounts from '@libs/actions/BankAccounts';
+import getClickedTargetLocation from '@libs/getClickedTargetLocation';
+import Log from '@libs/Log';
+import Navigation from '@libs/Navigation/Navigation';
+import * as PaymentUtils from '@libs/PaymentUtils';
+import * as ReportUtils from '@libs/ReportUtils';
+import * as PaymentMethods from '@userActions/PaymentMethods';
+import * as Policy from '@userActions/Policy';
+import * as Wallet from '@userActions/Wallet';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+import {BankAccountList, FundList, ReimbursementAccount, Report, UserWallet, WalletTerms} from '@src/types/onyx';
+import {AnchorPosition, DOMRectProperties, KYCWallProps, PaymentMethod, TransferMethod} from './types';
+
+// This sets the Horizontal anchor position offset for POPOVER MENU.
+const POPOVER_MENU_ANCHOR_POSITION_HORIZONTAL_OFFSET = 20;
+
+type BaseKYCWallOnyxProps = {
+    userWallet: OnyxEntry<UserWallet>;
+    walletTerms: OnyxEntry<WalletTerms>;
+    fundList: OnyxEntry<FundList>;
+    bankAccountList: OnyxEntry<BankAccountList>;
+    reimbursementAccount: OnyxEntry<ReimbursementAccount>;
+    chatReport: OnyxEntry<Report>;
+};
+
+type BaseKYCWallProps = KYCWallProps & BaseKYCWallOnyxProps;
+
+// This component allows us to block various actions by forcing the user to first add a default payment method and successfully make it through our Know Your Customer flow
+// before continuing to take whatever action they originally intended to take. It requires a button as a child and a native event so we can get the coordinates and use it
+// to render the AddPaymentMethodMenu in the correct location.
+function KYCWall({
+    addBankAccountRoute,
+    addDebitCardRoute,
+    anchorAlignment = {
+        horizontal: CONST.MODAL.ANCHOR_ORIGIN_HORIZONTAL.LEFT,
+        vertical: CONST.MODAL.ANCHOR_ORIGIN_VERTICAL.BOTTOM,
+    },
+    bankAccountList = {},
+    chatReportID = '',
+    children,
+    enablePaymentsRoute,
+    fundList,
+    iouReport,
+    onSelectPaymentMethod = () => {},
+    onSuccessfulKYC,
+    reimbursementAccount,
+    shouldIncludeDebitCard = true,
+    shouldListenForResize = false,
+    source,
+    userWallet,
+    walletTerms,
+    shouldShowPersonalBankAccountOption = false,
+}: KYCWallProps) {
+    const anchorRef = useRef(null);
```

Comment:
Add type for `anchorRef`, considering how it will be used (in which components)

Reply:
As I can see in one of the PR anchorRef is [typed as](https://github.com/Expensify/App/pull/33242/files#diff-52cad1bd9d75e9f709334465fb0ff19f0d0b09e9fda5faffec28b5c547c65110R71) `View | HTMLDivElement`.
Maybe you can go with `HTMLDivElement`
```
const anchorRef = useRef<HTMLDivElement>(null);
const transferBalanceButtonRef = useRef<HTMLDivElement | null>(null);
```